src/hotspot/share/opto/memnode.cpp

```diff
@@ -533,7 +533,23 @@ bool MemNode::detect_ptr_independence(Node* p1, AllocateNode* a1,
 Node* LoadNode::find_previous_arraycopy(PhaseTransform* phase, Node* ld_alloc, Node*& mem, bool can_see_stored_value) const {
   ArrayCopyNode* ac = find_array_copy_clone(phase, ld_alloc, mem);
   if (ac != NULL) {
-    return ac;
+    Node* ld_addp = in(MemNode::Address);
+    Node* ld_offs = ld_addp->in(AddPNode::Offset);
+    Node* src = ac->in(ArrayCopyNode::Src);
+    const TypeAryPtr* ary_t = phase->type(src)->isa_aryptr();
+
+    if (ary_t != NULL) {
```

Comment:
Needs Comment explaining what code does.

Reply:
Ok, I've added a few lines of comment. Let me know if you think something's still missing.